catalog/core/catalog-core-api/src/main/java/ddf/catalog/data/taxonomy/MetacardComponent.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.data.taxonomy;
+
+import java.util.Set;
+
+import ddf.catalog.data.AttributeDescriptor;
+
+/**
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ * </p>
+ */
+public interface MetacardComponent {
```

Comment:
We do not need this.  Instead the MetacardTypeImpl should have a constructor that can take a list of other `MetacardType`s and it will use the order of the types to break ties.  This is similar to how Java handles default methods.  It can also enforce there are no duplicate attribute names since the equals on an AttributeDescriptor could allow duplicate attribute descriptors with the same name.